src/main/java/uk/gov/hmcts/reform/pbis/servicebus/ServiceBusClient.java

```diff
@@ -0,0 +1,41 @@
+package uk.gov.hmcts.reform.pbis.servicebus;
+
+import com.microsoft.azure.servicebus.IMessage;
+import com.microsoft.azure.servicebus.IMessageReceiver;
+import java.util.UUID;
+
+
+public class ServiceBusClient implements IServiceBusClient {
+
+    private final IMessageReceiver messageReceiver;
+
+    public ServiceBusClient(IMessageReceiver messageReceiver) {
+        this.messageReceiver = messageReceiver;
+    }
+
+    @Override
+    public IMessage receiveMessage() {
+        try {
+            return this.messageReceiver.receive();
+        } catch (Exception ex) {
+            throw new ServiceBusException("Failed to receive message from subscription", ex);
+        }
+    }
+
+    @Override
+    public void completeMessage(UUID messageLockToken) {
+        try {
+            messageReceiver.complete(messageLockToken);
+        } catch (Exception e) {
+            throw new ServiceBusException(
+                String.format("Failed to mark message as completed", messageLockToken),
```

Comment:
missing `%s`?

Reply:
Good spot, I'll correct it

Reply:
Done. I'm including messageId in the message now.